keanu-project/src/test/java/io/improbable/keanu/vertices/LogProbGraphContract.java

```diff
@@ -11,4 +14,13 @@ public static void matchesKnownLogDensity(LogProbGraph logProbGraph, double expe
         double actualDensity = logProbGraphOutput.getValue().sum();
         assertEquals(expectedLogDensity, actualDensity, 1e-5);
     }
+
+    public static void equal(LogProbGraph actual, LogProbGraph expected) {
+        assertThat(actual.getLogProbOutput().getValue(), equalTo(expected.getLogProbOutput().getValue()));
+    }
+
+    public static void equalFlatArray(LogProbGraph logProbGraph, double[] flatArray) {
+        DoubleVertex logProbGraphOutput = logProbGraph.getLogProbOutput();
+        assertArrayEquals(logProbGraphOutput.getValue().asFlatDoubleArray(), flatArray, 1e-5);
+    }
```

Comment:
This kind of thing is more neatly expressed as a custom hamcrest matcher.

See `TensorMatchers` for some examples.